core/base/localizedTopologicalSimplification/LocalizedTopologicalSimplification.h

```diff
@@ -624,7 +624,7 @@ namespace ttk {
 
         segment.resize(propagation->segmentSize);
         IT segmentIndex = 0;
-        {
+        if(propagation->segmentSize > 0) {
```

Comment:
can `segmentSize` be really smaller than 0?

Reply:
(note the above resize)

Reply:
Well, GCC seemed to think that `propagation->SegmentSize == 0` might be a possible value. This would segfault when dereferencing the queue at line 633.

Reply:
this variable is intialized at zero in `core/base/localizedTopologicalSimplification/Propagation.h` and it is only incremented in both files.
I would vote to go against the insertion of that if statement, which looks unnecessary and may have an impact on performances.
@JonasLukasczyk what's your opinion?

Reply:
Segmentation size is initialized as 0, but theoretically after all propagation computations any segment must have a size of least 1. So 0 can only happen if the propagation procedure is buggy.

Reply:
A much more likely error scenario is when LTS is launched on a non simply connected manifold. This will crash for sure. But checking connectivity before LTS is a huge task.

Reply:
Maybe this issue can be detected during the propagation procedure, but this is not happening right now. I would say I can look into this when I have time, but I'm pretty busy right now.

Reply:
thanks @JonasLukasczyk for your quick input :)
let's keep the if statement then.